src/main/java/net/neoforged/neoforge/server/command/generation/GenerationTask.java

```diff
@@ -96,7 +96,8 @@ public void run(Listener listener) {
 
         this.listener = listener;
 
-        this.server.submit(() -> CompletableFuture.runAsync(this::tryEnqueueTasks));
+        // Off thread chunk scanning to skip already generated chunks
+        CompletableFuture.runAsync(this::tryEnqueueTasks);
```

Comment:
Calling `runAsync` without an explicit executor puts it on the common fork join pool. Is that intended? Might be better to explicitly pass Mojang's worker executor from `Util` instead.

(This was also an issue with the original code, but I didn't notice it then.)

Reply:
Can you explain more on what the differences and what end impact is?

Reply:
One is a worker pool created by Java at VM startup; it doesn't always have the right classloader associated with it, which has caused issues in the past on Forge. The other is created by Mojang and is what the rest of the chunk system uses.

Reply:
this? `CompletableFuture.runAsync(this::tryEnqueueTasks, Util.backgroundExecutor());`

Reply:
yes